File: javascript/core/src/verbs/boolean.ts
/*!
 * Copyright (c) Microsoft. All rights reserved.
 * Licensed under the MIT license. See LICENSE file in the project.
 */
import type { BooleanArgs } from '@datashaper/schema'

import { deriveBoolean } from './util/expressions.js'
import type { ColumnTableStep } from './util/factories.js'
import { stepVerbFactory } from './util/factories.js'

export const booleanStep: ColumnTableStep<BooleanArgs> = (
	input,
	{ columns = [], operator, to },
) => input.derive({ [to]: deriveBoolean(columns, operator) })

export const boolean = stepVerbFactory(booleanStep)

File: python/data-wrangling-components/data_wrangling_components/engine/verbs/boolean.py
#
# Copyright (c) Microsoft. All rights reserved.
# Licensed under the MIT license. See LICENSE file in the project.
#

from typing import List

from data_wrangling_components.engine.pandas.filter_df import _boolean_function_map
from data_wrangling_components.table_store import TableContainer
from data_wrangling_components.types import BooleanLogicalOperator


def boolean(input: TableContainer, to: str, columns: List[str], operator: str):

    logical_operator = BooleanLogicalOperator(operator)
    input_table = input.table
    output = input_table.copy()
    output[to] = _boolean_function_map[logical_operator](output)

    return TableContainer(table=output)
